### PR TITLE
feat: Minor updates and docker-compose compatibility

### PR DESCRIPTION
**Pull request by carlthelooper**
## Description and Motivation

Added some minor tweaks to package.json commands and added docker-compose compatibility. 

- Update to package.json commands (clean, prebuild)
- Update to dev dependencies
- Update to dockerfile (optimization)
- Addition of docker-compose including reading in .env
- Update to botConfig to allow for more robust parsing of pools and factory/router mapping envs

Note: suggested way of viewing or dumping the bot stdout logging output is to use the docker logs command from the host.  Example: `docker logs white-whale-bot > log.txt`

## Related Issues

---
## Checklist:

- [x] I have read [Migaloo's contribution guidelines](https://github.com/White-Whale-Defi-Platform/migaloo-bots/blob/main/CONTRIBUTING.md).
- [x] My pull request has a sound title and description (not something vague like `Update index.md`)
- [x] All existing and new tests are passing (if any).
- [x] I updated/added relevant documentation.
- [x] The code is formatted properly using eslint

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
+COPY tsconfig.json ./
+COPY src ./src
 RUN npm run build
-RUN mkdir /logs
-
-VOLUME ["/logs:/logs"]
-CMD ["node", "out/index.js", " > ", "logs/log.txt"]
 
+RUN mkdir /logs
+VOLUME ["/logs"]
```
Comment (SirTLB) [inline Dockerfile:16]: would this mean we are able to check console logs outside of the container? that is something that i want to add for a while

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+version: '3.9'
+
+services:
+  bot:
+    build: .
+    image: white-whale-bot
+    container_name: white-whale-bot
+    restart: always
+    stdin_open: true
+    tty: true
```
Comment (SirTLB) [inline docker-compose.yml:10]: can we add the environment file in here?

**Comment by SirTLB** [comment]
@carlthelooper could you maybe change this pr to release? thanks

File: Dockerfile (hunk 1 above)
Comment (carlthelooper) [reply Dockerfile:16]: It is the start, still need to do some mounting and such to actually get it to work.  I was interested in that as well and was going to get that work.  This PR is just the basics.

File: docker-compose.yml (hunk 2 above)
Comment (carlthelooper) [reply docker-compose.yml:10]: Yes, but docker-compose limits env variables to a single line so some changes in how pools are read need to happen first. Ideally, we'd just read in the env variables here so they could be altered on a per container basis and not copied into the docker image.  At that point, y'all could publish a docker image on docker hub and bots could be spun up without the need to download src.

**Comment by carlthelooper** [comment]
Actually almost have the fix to allow the docker-compose to read in the .env file.  I'll post a new PR against release with the fix shortly.

File: Dockerfile (hunk 1 above)
Comment (carlthelooper) [reply Dockerfile]: Seems like it is just easier to just dump into a file with the docker logs command. Mounting is just too finicky IMHO.  Example: `docker logs white-whale-bot > log.txt`

**Pull request closed without merging**